api/api/validator.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2015-2019, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+
+import json
+from xml.etree import ElementTree as ET
+import re
+
+
+regex_dict = {'numbers': r'^\d+$',
+              'array_numbers': r'^\d+(,\d+)*$',
+              'names': r'^[\w\-\.]+$',
+              'array_names': r'^[\w\-\.]+(,[\w\-\.]+)*$',
+              'paths': r'^[\w\-\.\\\/:]+$',
+              'dates': r'^\d{8}$',
+              'ips': r'^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(?:\/(?:[0-9]|[1-2][0-9]|3[0-2])){0,1}$|^any$|^ANY$',
+              'alphanumeric_param': r'^[\w,\-\.\+\s\:]+$',
+              'sort_param': r'^[\w_\-\,\s\+\.]+$',
+              'search_param': r'^[^;\|&\^*>]+$',
+              'select_param': r'^[\w\,\.]+$',
+              'ranges': r'[\d]+$|^[\d]{1,2}\-[\d]{1,2}$',
+              'hashes': r'^[\da-fA-F]{32}(?:[\da-fA-F]{8})?$|(?:[\da-fA-F]{32})?$',
+              'ossec_key': r'[a-zA-Z0-9]+$',
+              'timeframe_type': r'^(\d{1,}[d|h|m|s]?){1}$',
+              'empty_boolean': r'^$|(^true$|^false$)',
+              'yes_no_boolean': r'^yes$|^no$',
+              'boolean': r'^true$|^false$',
+              'query_param': r'[\w ]+$',
+              'type_format': r'^xml$|^json$',
+              'relative_paths': r'(^etc\/ossec.conf$)|((^etc\/rules\/|^etc\/decoders\/)[\w\-\/]+\.{1}xml$|(^etc\/lists\/)[\w\-\.\/]+)$'
+             }
+
+
+def check_params(parameters, filters):
+    """
+    Function to check multiple parameters
+    :param parameters: Dictionary with parameters to be checked
+    :param filters: Dictionary with filters for checking parameters
+    :return: True if parameters are OK, False otherwise
+    """
+    for key in parameters.keys():
+        if key not in filters.keys() or not check_exp(parameters[key], filters[key]):
+            return False
+
+    return True
+
+
+def check_exp(exp, regex_name):
+    """
+    Function to check if an expression matches a regex
+    :param exp: Expression to check
+    :param regex_name: Name of regular expression to do the matching
+    :return: True if expression is matched, False otherwise
+    """
+    return True if re.match(regex_dict[regex_name], exp) else False
+
+
+def check_path(relative_path):
```

Comment:
Maybe we could use a function like the one proposed [here](https://security.openstack.org/guidelines/dg_using-file-paths.html) to better prevent from path traversal attack, instead of simply filtering '.' or '..', which could be correct under some circumstances.

Reply:
Proposed function needs two parameters: `basedir` and `path`. We only are considering `path` for checking paths.

Reply:
Done!

Reply:
Well, we could safely suppose that `basedir` is always wazuh installation path by default. I would add that parameter as an optional kwarg for further use cases.